api/request.go

```diff
@@ -39,6 +43,20 @@ type FenceRequest struct {
 	Interpolation string `json:"interpolation" example:"linear"`
 } //@name FenceRequest
 
+/** Compute a hash of the request that uniquely identifies the requested fence
+ *
+ * The hash is computed based on all fields that contributes toward a unique response.
+ * I.e. very field except the sas token.
+ */
+func (f *FenceRequest) Hash() (string, error) {
```

Comment:
The point here was indeed to avoid a copy of fence. But maybe that is premature optimisation. I pass `f` to `cache.Hash`, and as you pointed out already `f` is `*FenceRequest` not `FenceRequest`. I.e. `cache.Hash` receives a pointer by value.

After reading [this](https://medium.com/@meeusdylan/when-to-use-pointers-in-go-44c15fe04eac) I'm thinking maybe you are right, and we should define `hash()` on the type not the pointer. Then we have a copy in this method and we can simply / safely change the copy before we send it off to `cache.Hash()`.  Even though that requires 2 extra copies